src/main.rs

```diff
@@ -1076,6 +965,138 @@ pub async fn main() -> anyhow::Result<()> {
     });
 }
 
+fn get_rainbow_color(start_time: Instant) -> [u8; 3] {
+    Hsva {
+        h: (start_time.elapsed().as_secs_f32() * 0.30) % 1.0,
+        s: 1.0,
+        v: 1.0,
+        a: 1.0,
+    }
+    .to_srgb()
+}
+
+fn add_ruler(
+    debugshapes: &mut core::cell::RefMut<'_, DebugShapes>,
```

Comment:
RefMut is not necessary here, `&mut DebugShapes` IIRC Refs are automatically coerced into normal references